src/ImageOptimizationLambda.ts

```diff
@@ -0,0 +1,113 @@
+import { Duration } from 'aws-cdk-lib';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { Function, FunctionOptions } from 'aws-cdk-lib/aws-lambda';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import * as s3Assets from 'aws-cdk-lib/aws-s3-assets';
+
+import { Construct } from 'constructs';
+import * as fs from 'fs-extra';
+import { bundleFunction } from './BundleFunction';
+import { NextjsBaseProps } from './NextjsBase';
+import { createArchive, NextjsBuild } from './NextjsBuild';
+import { NextjsLayer } from './NextjsLayer';
+
+
+function getEnvironment(props: ImageOptimizationProps): { [name: string]: string } {
+  const environmentVariables: { [name: string]: string } = {
+    ...props.environment,
+    ...props.function?.environment,
+    ...(props.nodeEnv ? { NODE_ENV: props.nodeEnv } : {}),
+  };
+
+  return environmentVariables;
+}
+
+export interface ImageOptimizationProps extends NextjsBaseProps {
+  /**
+   * The internal S3 bucket for application images.
+   */
+  bucketName: string;
+  /**
+   * Built nextJS application.
+   */
+  readonly nextBuild: NextjsBuild;
+
+  /**
+   * Override function properties.
+   */
+  readonly function?: FunctionOptions;
+}
+
+const RUNTIME = lambda.Runtime.NODEJS_16_X;
+
+/**
+ * This lambda handles image optimization.
+ */
+export class ImageOptimizationLambda extends Construct {
+  configBucket?: Bucket;
+  lambdaFunction: Function;
+
+  constructor(scope: Construct, id: string, props: ImageOptimizationProps) {
+    super(scope, id);
+    const { nextBuild, function: functionOptions, bucketName } = props;
+
+    // build our server handler in build.nextStandaloneDir
+    const serverHandler = path.resolve(__dirname, '../assets/lambda/ImageOptimization.ts');
+    // server should live in the same dir as the nextjs app to access deps properly
+    const serverPath = path.join(props.nextjsPath, 'server.cjs');
+    bundleFunction({
+      inputPath: serverHandler,
+      outputPath: path.join(nextBuild.nextStandaloneDir, serverPath),
+      bundleOptions: {
+        bundle: true,
+        minify: false,
+        sourcemap: true,
+        target: 'node16',
+        platform: 'node',
+        external: ['sharp', 'next', 'aws-sdk'],
+        format: 'cjs', // hope one day we can use esm
+      },
+    });
+
+    // zip up the standalone directory
+    const zipOutDir = path.resolve(
+      props.tempBuildDir
+        ? path.resolve(path.join(props.tempBuildDir, `imageOptimization`))
+        : fs.mkdtempSync(path.join(os.tmpdir(), 'imageOptimization-'))
+    );
+    const zipFilePath = createArchive({
+      directory: nextBuild.nextStandaloneDir,
+      zipFileName: 'imageOptimization.zip',
+      zipOutDir,
+      fileGlob: '*',
+      quiet: props.quiet,
+    });
+    
+    if (!zipFilePath) throw new Error('Failed to create archive for image optimization lambda function code');
+
+    // build native deps layer
+    const nextLayer = new NextjsLayer(scope, 'ImageOptimizationLayer', {});
```

Comment:
Not sure how to reuse this layer, the only diff is the `scope`. Does CDK allow you to pass in an `id` in the 3rd parameter to reuse an already declared resource?